libbeat/processors/actions/rename_fields_test.go

```diff
@@ -0,0 +1,96 @@
+package actions
+
+import (
+	"testing"
+
+	"github.com/ruflin/docker/pkg/testutil/assert"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func TestRenameRun(t *testing.T) {
+	var tests = []struct {
+		Fields        map[string]string
+		IgnoreMissing bool
+		Input         common.MapStr
+		Output        common.MapStr
+		error         bool
+	}{
+		{
+			Fields: map[string]string{
+				"a": "c",
+			},
+			Input: common.MapStr{
+				"a": "b",
+			},
+			Output: common.MapStr{
+				"c": "b",
+			},
+			error: false,
+		},
+		{
+			Fields: map[string]string{
+				"a.b": "a.b.c",
+			},
+			Input: common.MapStr{
+				"a.b": 12,
+			},
+			Output: common.MapStr{
+				"a": common.MapStr{
+					"b": common.MapStr{
+						"c": 12,
+					},
+				},
+			},
+			error: false,
+		},
+		{
+			// This overwrites an existing field
```

Comment:
@andrewkroh @ph I wonder what our expected behaviour here is. We could later add an option "prevent overwrite" or something if it becomes an issue.

Reply:
Well, I think we should aim for the **least surprise**, what I mean, is we should fails if the target field already exist. They can always use a `drop_field` before a rename to ensure it always works?

Reference:
https://github.com/elastic/elasticsearch/blob/master/modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/RenameProcessor.java#L68-L75